### PR TITLE
Add makeblastdb wrapper

### PR DESCRIPTION
**Pull request by nimne**
This pull request addresses issue #1788, with the purpose of adding a makeblastdb wrapper.

In this pull request I have added ncbi-blast+ to the TravisCI checks. It doesn't seem to add more than a few seconds to the build time, but I'm happy to turn these off if the build time starts to get too high.

When I added the `NcbimakeblastdbCommandline` class, I didn't subclass from `_NcbibaseblastCommandline` because so few of the command line arguments were the same between them. The other option would have been to split `_NcbibaseblastCommandline` into another superseding class with only `-h` `-help` `-version` and `-out`. I thought this was a little more convoluted than not subclassing at all but it doesn't fit with the existing style as nicely, I'm happy to change this if needed.

This should show up in the generated docs, but is there anywhere else that this should be documented? 

- [X] I hereby agree to dual licence this and any previous contributions under both
the _Biopython License Agreement_ **AND** the _BSD 3-Clause License_.

- [X] I have read the ``CONTRIBUTING.rst`` file and understand that AppVeyor and
TravisCI will be used to confirm the Biopython unit tests and ``flake8`` style
checks pass with these changes.

- [X] I have added my name to the alphabetical contributors listings in the files
``NEWS.rst`` and ``CONTRIB.rst`` as part of this pull request, am listed
already, or do not wish to be listed. (*This acknowledgement is optional.*)

### CODE REVIEW

**Comment by nimne** [comment]
Looks like the codecov report showed some deficiencies. I don't understand the `Bio/GA/Evolver.py` or `Bio/Graphics/BasicChromosome.py` changes, but I can make the `Bio/Blast/Applications.py` coverage better.

**Comment by peterjc** [comment]
Sadly we do have some stochastic test coverage in the GA code and some of the graphics (the later should be easy to fix - rather than a random list for the scatter plot, just hard code it), and as result the CodeCov numbers are often a tiny fraction negative. This is one reason why we don't require the CodeCov checks to pass.

**Comment by peterjc** [comment]
Looking at your coverage output, I'd be happy with it as it is.

File: Bio/Blast/Applications.py [hunk 1]
```diff
@@ -1331,13 +1338,21 @@ def __init__(self, cmd="makeblastdb", **kwargs):
...
         ]
-        _NcbibaseblastCommandline.__init__(self, cmd, **kwargs)
+        AbstractCommandline.__init__(self, cmd, **kwargs)
+
+    def _input_type_checker(command, x):
+        expression = (x == 'asn1_bin' or
+                      x == 'asn1_txt' or
+                      x == 'blastdb' or
+                      x == 'fasta')
+        return expression
```
Comment (peterjc) [inline Bio/Blast/Applications.py]: I see you've only moved this old code, but I think we should simplify it to just:

```python
return x in ('asn1_bin', 'asn1_txt', 'blastdb', 'fasta')
```

**Comment by peterjc** [comment]
I am a bit surprised BLAST+ wasn't already being installed on TravisCI - I thought I would have tried that long ago, but maybe not? It does not seem to affect the build run time much, so let's leave it in.

Minor comment on a helper function, otherwise this looks ready to merge. Thank you.

File: Bio/Blast/Applications.py (hunk 1 above)
Comment (nimne) [reply Bio/Blast/Applications.py]: This is much simpler! I like the change.

**Comment by peterjc** [comment]
Merged, thank you!

**Pull request merged**